### PR TITLE
Add the possibility to edit the name of a saved file in the inspector…

### PR DESCRIPTION
**Pull request by francoisph**
… view

### CODE REVIEW

File: client/src/model/SaveFile.js [hunk 1]
```diff
@@ -15,9 +15,10 @@ define( [
 
     var SaveFile = ScenicModel.extend( {
         defaults:    {
-            name:    null,
-            date:    null,
-            current: false
+            name:       null,
+            editName:   false,
```
Comment (ubald) [inline client/src/model/SaveFile.js]: ```editName``` should be in the view only, it is not related to the model.

File: server/src/scenic/SwitcherController.js [hunk 2]
```diff
@@ -386,6 +386,36 @@ class SwitcherController {
...
+        var newFilePath = this.config.savePath + newfileName + '.json';
+
+        var self = this;
+        try {
+            fs.rename( oldFilePath, newFilePath, function ( error ) {
+                if ( error ) {
+                    log.error( error );
+                    return cb( error );
+                }
+                log.info( 'Scenic file renameFile', newFilePath );
```
Comment (ubald) [inline server/src/scenic/SwitcherController.js]: renamed ;)

File: client/src/model/SaveFile.js [hunk 3]
```diff
@@ -52,7 +54,25 @@ define( [
...
+        _onFileRenamed( result ) {
+            if ( result.filename == this.get( 'name' )) {
+                this.set('name',  result.rename );
+            }
+        },
+
+        renameFile: function ( name ) {
+            this.scenic.socket.emit( 'file.rename', this.get( 'name' ), name, function ( error) {
+                if ( error ) {
+                    console.log('rename File error ', error)
```
Comment (ubald) [inline client/src/model/SaveFile.js]: You should emit an error here:
```
this.scenic.sessionChannel.vent.trigger( 'error', error );
```
Of course, to use ```this``` you'll have to change the callback to an arrow function ``` error => { ... }```

**Comment by ubald** [comment]
Edit button should be aligned with the remove button and feature a blue colored highlight like the red one for the remove button.
screenshot from 2016-08-15 13-31-55

**Comment by ubald** [comment]
@francoisph Reviewed this PR.

**Comment by francoisph** [comment]
@ubald modification review

**Pull request merged**